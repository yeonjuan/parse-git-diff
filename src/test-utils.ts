import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
