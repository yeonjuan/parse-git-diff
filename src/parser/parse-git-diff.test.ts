import { createContext } from '../test-utils';
import parseGitDiff from './parse-git-diff';

describe('parseGitDiff', () => {
  it('new file.', () => {
    // prettier-ignore
    const src =
  `diff --git a/parse-git-diff-test/packages.json b/parse-git-diff-test/packages.json
  new file mode 100644
  index 0000000..5515040
  --- /dev/null
  +++ b/parse-git-diff-test/packages.json
  @@ -0,0 +1,17 @@
  +{
  +  "name": "parse-git-diff-test",
  +  "version": "1.0.0",
  +  "description": "",
  +  "main": "index.js",
  +  "scripts": {
  +    "build": "tsc"
  +  },
  +  "author": "",
  +  "license": "ISC",
  +  "dependencies": {
  +    "parse-git-diff": "0.0.3"
  +  },
  +  "devDependencies": {
  +    "typescript": "^4.4.4"
  +  }
  +}`;

    const result = parseGitDiff(src);

    expect(result).not.toBe(null);
    expect(result).toMatchSnapshot();
  });

  it('deleted file', () => {
    // prettier-ignore
    const src =
    `diff --git a/parse-git-diff-test/package.json b/parse-git-diff-test/package.json
  deleted file mode 100644
  index 5515040..0000000
  --- a/parse-git-diff-test/package.json
  +++ /dev/null
  @@ -1,17 +0,0 @@
  -{
  -  "name": "parse-git-diff-test",
  -  "version": "1.0.0",
  -  "description": "",
  -  "main": "index.js",
  -  "scripts": {
  -    "build": "tsc"
  -  },
  -  "author": "",
  -  "license": "ISC",
  -  "dependencies": {
  -    "parse-git-diff": "0.0.3"
  -  },
  -  "devDependencies": {
  -    "typescript": "^4.4.4"
  -  }
  -}`;
    const result = parseGitDiff(src);

    expect(result).not.toBe(null);
    expect(result).toMatchSnapshot();
  });

  it('rename', () => {
    // prettier-ignore
    const src =
`diff --git a/parse-git-diff-test/tsconfig.json b/parse-git-diff-test/ts.json
similarity index 100%
rename from parse-git-diff-test/tsconfig.json
rename to parse-git-diff-test/ts.json`;
    const result = parseGitDiff(src);
    expect(result).not.toBe(null);
    expect(result).toMatchSnapshot();
  });
});
