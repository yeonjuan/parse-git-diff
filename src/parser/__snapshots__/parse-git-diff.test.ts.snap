// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseGitDiff deleted file 1`] = `
Object {
  "files": Array [],
  "type": "GitDiff",
}
`;

exports[`parseGitDiff new file. 1`] = `
Object {
  "files": Array [],
  "type": "GitDiff",
}
`;

exports[`parseGitDiff rename 1`] = `
Object {
  "files": Array [
    Object {
      "chunks": Array [],
      "pathAfter": "parse-git-diff-test/ts.json",
      "pathBefore": "parse-git-diff-test/tsconfig.json",
      "type": "RenamedFile",
    },
  ],
  "type": "GitDiff",
}
`;
